shared/common-adapters/clickable-box.js.flow

```diff
@@ -5,7 +5,7 @@ export type Props = {
   className?: ?string,
   children?: any,
   style?: any,
-  onClick?: ?(event: SyntheticEvent<>) => void,
+  onClick?: ?(event: SyntheticEvent<>) => any,
```

Comment:
why return any?

Reply:
I was trying to use this:
```
<ClickableBox onClick={evt => !evt.defaultPrevented && onClick()} ... >
```
and flow complained about `boolean incompatible with undefined` in the return value. It seems weird to have to change it for the adapter that doesn't actually care about what is returned, so I think `any` makes more sense as a stand in for `don't care`.

Reply:
I see...does keeping `onClick` returning `void` and having:

`onClick={evt => { !evt.defaultPrevented && onClick() }}`

work?

Reply:
Factoring out those functions in `header.js` avoids the problem. Changed it back, I still wonder whether `void` is the right type for those, but it doesn't matter much.